Refactor dir

gbwlxhd97 opened the pull request:
## 🎫 [지라 티켓 번호] - 
- 작은 단위라 지라를 통해서 만들지 않았음
<br>

<br>

## 🛠️ 작업 내용
<!-- 작업 내용 (스크린샷도 같이 있으면 좋아요) -->
![image](https://github.com/PBTP/Webview/assets/49021925/5dc1c1b1-74e1-4f65-a840-a703e2afbeb0)

<br>
1. Reservation페이지 컴포넌트를 만들어서 
그동안 작업했던 캘린더 컴포넌트와 버튼 컴포넌트를 붙여서 스크린샷처럼 만들었습니다.
2. 폴더 이름을 좀 변경하였습니다.
예시로 /pages/Reservation 폴더까지 붙어 있기 떄문에 DetailReservation -> Detail로 변경하였습니다.
첫번째 예약 페이지를 Index, 두번째 예약 페이지를 detail로 네이밍 하였습니다.
<br>

## 🗒️ Note (optional)
<!-- 추가 필요한 사항이나 하고픈 말
     Reviewer 한테 요청하고 싶은 것들
     코드리뷰 요청하고 싶은 것들.. 등등 -->

gbwlxhd97 committed:
refactor : dir
fix : calendar css

gbwlxhd97 committed:
Merge branch 'main' of https://github.com/PBTP/Webview into refactor-dir

gbwlxhd97 committed:
1. refactor: reservation page dir update
2. reservation page complete

VictoryJu commented on src/components/common/Calendar/Calendar.tsx:
```diff
@@ -38,10 +38,20 @@ const CMCalendar = () => {
         tileClassName={tileClassName}
         prevLabel={
-          <ArrowIcon width={20} height={20} stroke="#8B95A1" fill="none" />
+          <ArrowIcon
```
이친구 파일이랑 컴포넌트 이름 ArrowRightIcon 으로 재정의 하시는 것 어떤가요?? 기본이 어느 방향으로 보이는지 헷갈릴것같아서용 

fill은 사용안되는것같아서 props에서 제거해도 될것같습니다.

gbwlxhd97 replied:
넵 이부분 수정하겠습니다.

gbwlxhd97 replied:
흠 제가 svg를 잘못 추출 한 걸까요?
fill="none"을 작성하지 않으면 아래 사진처럼 됩니다!
왼쪽이 fill="none" 하지 않은 사진
오른쪽이 fill="none"한 사진
![image](https://github.com/PBTP/Webview/assets/49021925/28112cab-97e2-436f-a3dc-afd61562c797)

gbwlxhd97 replied:
ArrowIcon은 네이밍 변경하겠습니다!

VictoryJu replied:
@gbwlxhd97 오우 화살표만드는 방식이 검정부분을 fill="none"으로 지우는것같네요 
props자체를 제거하면 될듯싶습니다.

ArrowIcon 컴포넌트를 사용할때 fill="none"을 props로 안넘기면 undefined가 들어가서 아마 fill="undefined"가 됐을것같네요 fill="none"이여야 화살표가 지워지는것같아요 그래서 안됐던거구요

파일자체에 "none"이 박혀있어서 props로 따로 안받고 stroke로만 색상 바꾸면 될듯싶습니다.

이렇게가시져
```typescript
const ArrowIcon = ({ width, height, stroke, className }: IconProps) => {
  return (
    <ArrowSvg
      className={className}
      width={width}
      height={height}
      stroke={stroke}
    />
  );
};
```

VictoryJu replied:
@gbwlxhd97 svg파일도 fill="current" 로 들어가있는것같아서 ArrowIcon SVG파일도 이거로 바꾸시져 

```html
<svg width="current" height="current" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.5 15L12.5 10L7.5 5" stroke="current" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
```

VictoryJu commented on src/components/common/Calendar/Calendar.tsx:
```diff
@@ -38,10 +38,20 @@ const CMCalendar = () => {
+            width={20}
+            height={20}
+            className={styles.NextArrowIcon}
+            fill="none"
```
여기두 fill안쓰면 삭제해도될것같습니다.

VictoryJu commented on src/pages/Reservation/Index/Index.module.scss:
```diff
@@ -0,0 +1,37 @@
+  margin-top: 16px;
+  > .TimeButton {
+    padding: 12px 14px;
+    flex-basis: calc(20% - 10px); /* 한 행에 최대 5개의 요소가 들어가도록 설정 */
```
10px 들어가야되나유?? gap이 없어보여서요

gbwlxhd97 replied:
10px 뺀 스크린샷
![image](https://github.com/PBTP/Webview/assets/49021925/d2b10d38-ec47-4c47-b454-eea34daa16b8)
-----
10px 그대로 넣은 스크린샷
![image](https://github.com/PBTP/Webview/assets/49021925/4b989dd8-d218-42f6-b8a1-60bdcf2f64d3)

이래 됩니다!

VictoryJu replied:
@gbwlxhd97  아 저게 gap 역할을 하고있었던거군요 

혹시 이 코드는 어떤가요??

grid로 gap주면서 한줄에 5개씩 들어가게 하는 코드입니다.
gap을 명시해서 좀더 직관적이게 만들어봤습니다

지원님 코드도 좋아서 취향대로 하시면될듯합니다.

```css
  display: grid;
  grid-template-columns: repeat(5, minmax(0, 1fr));
  gap: 10px;
```

gbwlxhd97 replied:
이건 이 코드 따라갈게요
사실 너무 오랜만이라 그리드를 생각을 못해서 gpt한테 시켰습니다 ㅎㅎ
저도 이 코드가 더 좋은 것 같아요

VictoryJu commented on src/pages/Reservation/Index/Index.tsx:
```diff
@@ -0,0 +1,67 @@
+      <div className={styles.Title}>예약 스케쥴</div>
+      <CMCalendar />
+      <div className={styles.DividerWrap}>
+        <div className="Divider" />
```
오우 margin이란 변수가있었군요

이러면 컴포넌트로 따로 빼거나 그냥 직접 만드는게 나을수도있겠는데요?
``` typescript
// Divier.tsx
import styles from './Divider.module.scss'

type DividerProps{
 className?: string
}
const Divider = ({className}:DividerProps) =>{
  return <div className={`${styles.DividerWrapper}  ${className}`} />
}

//Divider.module.scss
@import '@/styles/theme.scss'
.DividerWrapper{
  border: 1px solid $gray300;
}

//사용하는곳
//마진값만 따로주기
... <Divider className={styles.ReservationDivider} />
```

VictoryJu replied:
@gbwlxhd97 엇 방금 떠오른건데 컴포넌트 안만들고

```typescript
<div className={`${styles.ReservationDivider} Divider`}/>
```

이런식으로 className을 같이쓰면 될것같습니당ㅋㅋ 이렇게쓰시면 styles.DividerWrap으로 div를 하나 더 감싸셨는데 그걸 없앨수있을것같아요

VictoryJu reviewed (approved):
코멘트 반영되면 바로 머지해주셔도 될것같습니다!

gbwlxhd97 committed:
1. Refactor : Calendar.tsx divide utils function
2.ArrowIcon name update
before -> ArrowIcon , after -> ArrowRightIcon
3.create Divider global css

gbwlxhd97 committed:
1. fix : conflict solve

gbwlxhd97 commented:
@VictoryJu 
Calendar.tsx 내부 함수를 utils 폴더에 함수로 만들어서 컴포넌트와 함수 분리를 추가적으로 하였고
리뷰해주신 부분 참고해서 머지 하겠습니다~
